usaspending_api/common/etl/spark.py

```diff
@@ -560,3 +564,177 @@ def create_ref_temp_views(spark: SparkSession, create_broker_views: bool = False
             spark.sql(sql_statement)
 
     logger.info(f"Created the reference views in the global_temp database")
+
+
+def write_csv_file(
+    spark: SparkSession,
+    df: DataFrame,
+    parts_dir: str,
+    max_rows_per_merged_file=EXCEL_ROW_LIMIT,
+    overwrite=True,
+    logger=None,
+) -> int:
+    """Write DataFrame data to CSV file parts.
+    Args:
+        spark: passed-in active SparkSession
+        df: the DataFrame wrapping the data source to be dumped to CSV.
+        parts_dir: Path to dir that will contain the outputted parts files from partitions
+        overwrite: Whether to replace the file CSV files if they already exist by that name
+        max_rows_per_merged_file: Suggestion to Spark of how many records to put in each written CSV file part,
+        if it will end up writing multiple files.
+        logger: The logger to use. If one note provided (e.g. to log to console or stdout) the underlying JVM-based
+            Logger will be extracted from the ``spark`` ``SparkSession`` and used as the logger.
+    Returns:
+        record count of the DataFrame that was used to populate the CSV file(s)
+    """
+    if not logger:
+        logger = get_jvm_logger(spark)
+    # Delete output data dir if it already exists
+    parts_dir_path = spark.sparkContext._jvm.org.apache.hadoop.fs.Path(parts_dir)
+    fs = parts_dir_path.getFileSystem(spark.sparkContext._jsc.hadoopConfiguration())
+    if fs.exists(parts_dir_path):
+        fs.delete(parts_dir_path, True)
+    start = time.time()
+    logger.info(f"Writing source data DataFrame to csv part files for file {parts_dir}...")
+    df_record_count = df.count()
+    df.write.options(
+        # NOTE: this is a suggestion, to be used by Spark if partitions yield multiple files
+        maxRecordsPerFile=max_rows_per_merged_file,
+    ).csv(
+        path=parts_dir,
+        header=False,
+        nullValue=None,
+        escape='"',  # " is used to escape the 'quote' character setting (which defaults to "). Escaped quote = ""
+        ignoreLeadingWhiteSpace=False,  # must set for CSV write, as it defaults to true
+        ignoreTrailingWhiteSpace=False,  # must set for CSV write, as it defaults to true
+        timestampFormat=CONFIG.SPARK_CSV_TIMEZONE_FORMAT,
+        mode="overwrite" if overwrite else "errorifexists",
+    )
+    logger.info(f"{parts_dir} contains {df_record_count:,} rows of data")
+    logger.info(f"Wrote source data DataFrame to csv part files in {(time.time() - start):3f}s")
+    return df_record_count
+
+
+def hadoop_copy_merge(
+    spark: SparkSession,
+    parts_dir: str,
+    header: str,
+    max_rows_per_merged_file=EXCEL_ROW_LIMIT,
+    logger=None,
+    file_format=".csv",
```

Comment:
Might be a little tricky requiring the `.` as part of the `file_format`. See the Spark write's format as an example: https://sparkbyexamples.com/spark/spark-write-options/

Can you update the default value along with the instantiation of `merged_file_name`?